src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -122,16 +123,18 @@ export class CalciteRadioButton {
   //--------------------------------------------------------------------------
 
   private checkFirstRadioButton(): void {
-    const radioButtons = document.querySelectorAll(`calcite-radio-button[name=${this.name}]`);
+    const radioButtons = Array.from(document.querySelectorAll("calcite-radio-button")).filter(
+      (radioButton) => radioButton.name === this.name
+    ) as HTMLCalciteRadioButtonElement[];
     let firstCheckedRadioButton: HTMLCalciteRadioButtonElement;
     if (radioButtons && radioButtons.length > 0) {
       radioButtons.forEach((radioButton: HTMLCalciteRadioButtonElement) => {
         if (firstCheckedRadioButton) {
           radioButton.checked = false;
+          this.calciteRadioButtonCheckedChange.emit(false);
```

Comment:
Emitting this new `checkedChange` event here is what `tile-select` needs in order to update its styling appropriately when multiple `tile-selects` have their `checked` attribute set initially.